apps/storage/dbworker.cpp

```diff
@@ -489,6 +490,47 @@ void DBWorker::clearHistory()
     emit historyAvailable(linkList);
 }
 
+void DBWorker::clearHistoryWithinPeriod(int days)
+{
+    QList<Link> historyList = getHistoryQList();
+    QSet<QString> historyIconsToRemove;
+
+    int historySize = historyList.size();
+    int boundaryIndex = 0;
+
+    for (int i = 0; i < historySize; i++) {
+        const QString sanitizedUrl = FaviconManager::sanitizedHostname(historyList[i].url());
+        if (historyList[i].date().daysTo(QDate::currentDate()) < days && !historyIconsToRemove.contains(sanitizedUrl)) {
+            historyIconsToRemove.insert(sanitizedUrl);
+        } else if (historyList[i].date().daysTo(QDate::currentDate()) >= days) {
+            boundaryIndex = i;
+            break;
+        }
+        boundaryIndex = i + 1;
+    }
+
+    for (int i = boundaryIndex; i < historySize; i++) {
+        const QString sanitizedUrl = FaviconManager::sanitizedHostname(historyList[i].url());
+        if (historyIconsToRemove.contains(sanitizedUrl)) {
+            historyIconsToRemove.remove(sanitizedUrl);
+        }
+    }
+
+    QSet<QString>::iterator iconIterator;
+    for (iconIterator = historyIconsToRemove.begin(); iconIterator != historyIconsToRemove.end(); iconIterator++) {
+        FaviconManager::instance()->remove(QStringLiteral("history"), *iconIterator);
+    }
+
+    uint boundaryTimeT = QDateTime::currentDateTimeUtc().addDays(-days).toTime_t();
+
+    QSqlQuery query = prepare("DELETE FROM browser_history WHERE date > ?");
+    query.bindValue(0, boundaryTimeT);
+    execute(query);
+    removeAllTabs();
```

Comment:
This call is duplicated
https://github.com/sailfishos/sailfish-browser/pull/914/files#diff-6ee2a3c1bdc5d546b02a363a605dfbc2fbe3287b0adf62f0208b49cc9a29f5c6R184

Reply:
if (clearHistory.checked) {
                                                         Settings.clearHistory()
                                                         Settings.removeAllTabs()
                                                     }